### ISSUE TITLE
Bash completion

### ISSUE THREAD

**Issue by gotlium** [eddiezane/lunchy#46]
Can you add bash-completion support, for using Tab to autocomplete. That is very helpful when working on terminal.

**Comment by sammcj** [comment]
+1

**Comment by aslikeyou** [comment]
+1

**Comment by bunnymatic** [comment]
Here is a simple working completion file that i use

```
# completion-lunchy
__from_lunchy_list() {
  COMPREPLY=()
  local cur=${COMP_WORDS[COMP_CWORD]}
  COMPREPLY=($(compgen -W '$(lunchy list)' -- $cur))
}

complete -F __from_lunchy_list -o default lunchy
```

Simply save this file somewhere (like ~/bash_completions) and include in your `.profile` or `.bashrc` like

```
if [ -f ~/bash_completions/completion-lunchy ]; then
  . ~/bash_completions/completion-lunchy
fi
```

**Comment by alexeyshockov** [comment]
Here another variant that I use. Completion will work not only for services, but for actions (start, stop,..) too.

```
function _lunchy {
  COMPREPLY=()
  local cur=${COMP_WORDS[COMP_CWORD]}
  local cur_pos=$COMP_CWORD;

  case "$cur_pos" in
    1) COMPREPLY=($(compgen -W 'ls list start stop restart status install uninstall show edit' -- $cur))
       ;;
    *) COMPREPLY=($(compgen -W '$(lunchy list)' -- $cur))
       ;;
  esac
}

complete -F _lunchy -o default lunchy
```

You can simple save this to your `~/.bash_completion` file and it will work.

@eddiezane, what do you think about this?

**Comment by eddiezane** [comment]
These look great! Not quite sure what the practice is for shipping completion scripts with gems. Ideas?

**Comment by alexeyshockov** [comment]
You ship CLI command with gem. Why don't ship completion for it too?

On Wednesday, June 17, 2015, Eddie Zaneski <notifications@github.com> wrote:

> These look great! Not quite sure what the practice is for shipping
> completion scripts with gems. Ideas?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/eddiezane/lunchy/issues/46#issuecomment-112808767>.
>


-- 
Best regards,
Alexey Shockov

tel:+79652121992
skype:alexeyshockov
mailto:alexey@shockov.com
http://alexey.shockov.com/

**Comment by bunnymatic** [comment]
This gem just spits out instructions on what to copy where :
https://github.com/colszowka/rvm-completion/blob/master/lib/rvm-completion.rb

It seems like it could be as simple as that.  Like
```
Add the following to your .bashrc
if [ -f $GEM_PATH/lunchy/version/completion.bash ]; then
   $GEM_PATH/lunchy/version/completion.bash
fi
```

**Comment by eddiezane** [comment]
Thanks for the great collaboration all! Released in v0.10.0.

**Comment by bunnymatic** [comment]
love it.  just installed this on a new box and saw my post install comments!
